main.go

```diff
@@ -266,6 +270,73 @@ func CreateDeployment(c *gin.Context) {
 	return
 }
 
+type EndpointResponse struct {
+	Endpoints []*EndpointItem `json:"endpoints"`
+}
+
+type EndpointItem struct {
+	Name string `json:"name"`
+	Host string `json:"host"`
+}
+
+type ErrorResponse struct {
+	Code    int    `json:"code"`
+	Message string `json:"message"`
+}
+
+// List service public endpoints
+func ListEndpoints(c *gin.Context) {
+	var endpointList []*EndpointItem
+
+	// --- [ Get cluster ] ---- //
+	banzaiUtils.LogInfo(banzaiConstants.TagListDeployments, "Get cluster")
+	cloudCluster, err := cloud.GetClusterFromDB(c)
+	if err != nil {
+		return
+	}
+	banzaiUtils.LogInfo(banzaiConstants.TagListDeployments, "Getting cluster succeeded")
+
+	// --- [ Get K8S Config ] --- //
+	kubeConfig, err := cloud.GetK8SConfig(cloudCluster, c)
+	if err != nil {
+		return
+	}
+	apiconfig, _ := clientcmd.Load(kubeConfig)
+	clientConfig := clientcmd.NewDefaultClientConfig(*apiconfig, &clientcmd.ConfigOverrides{})
+	config, err := clientConfig.ClientConfig()
+	if err != nil {
+		banzaiUtils.LogError(banzaiConstants.TagKubernetes, "Could not create kubernetes client from config.")
+		c.JSON(http.StatusBadRequest, ErrorResponse{
+			Code:    http.StatusBadRequest,
+			Message: fmt.Sprintf("create kubernetes client failed: %v", err),
+		})
+		return
+	}
+	client, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		banzaiUtils.LogError(banzaiConstants.TagKubernetes, "Could not create kubernetes client from config.")
+		c.JSON(http.StatusBadRequest, ErrorResponse{
+			Code:    http.StatusBadRequest,
+			Message: fmt.Sprintf("create kubernetes client failed: %v", err),
+		})
+		return
+	}
+	serviceList, err := client.CoreV1().Services("").List(meta_v1.ListOptions{})
+	for _, service := range serviceList.Items {
+		banzaiUtils.LogDebugf(banzaiConstants.TagKubernetes, "Service: %#v", service.Status)
+		if len(service.Status.LoadBalancer.Ingress) > 0 {
+			endpointList = append(endpointList, &EndpointItem{
+				Name: service.Name,
+				Host: service.Status.LoadBalancer.Ingress[0].Hostname,
```

Comment:
Does this work AKS as well? Apparently on AKS the Azure LB created for LoadBalancer type Kubernetes services has a (static) IP instead contrary to AWS.

Reply:
No it does not works on AKS because there the IP is used instead of Hostname